File: frontend/src/types/race.ts
export interface CircuitOut {
  circuit_id: string;
  circuit_ref: string;
  name: string;
  location: string;
  country: string;
  lat?: number;
  lng?: number;
  url?: string;
}

export interface SessionBrief {
  session_id: number;
  session_type: string;
  date: string;
  time?: string;
  status: 'scheduled' | 'completed' | 'cancelled';
}

export interface RaceBrief {
  race_id: number;
  year: number;
  round: number;
  name: string;
  date: string;
  time?: string;
  circuit: CircuitOut;
  status: 'scheduled' | 'completed' | 'cancelled';
  winner_name?: string;
  pole_name?: string;
}

export interface RaceDetail extends RaceBrief {
  sessions: SessionBrief[];
  url?: string;
}

File: backend/app/schemas/race.py
from datetime import date as Date, time as Time

from pydantic import BaseModel

from app.schemas.session import SessionBrief


class CircuitOut(BaseModel):
    id: int
    circuit_ref: str
    name: str
    location: str
    country: str
    lat: float | None = None
    lng: float | None = None
    alt: int | None = None
    url: str | None = None

    class Config:
        from_attributes = True


class RaceBrief(BaseModel):
    id: int
    round: int
    name: str
    date: Date
    time: Time | None = None
    circuit_name: str
    circuit_country: str
    winner_name: str | None = None
    pole_name: str | None = None

    class Config:
        from_attributes = True


class RaceDetail(BaseModel):
    id: int
    round: int
    name: str
    official_name: str | None = None
    date: Date
    time: Time | None = None
    url: str | None = None
    circuit: CircuitOut
    sessions: list[SessionBrief] = []

    class Config:
        from_attributes = True
